packages/big-design/src/components/ProgressCircle/ProgressCircle.tsx

```diff
@@ -10,26 +12,10 @@ export interface ProgressCircleProps {
   size?: 'xSmall' | 'small' | 'medium' | 'large';
 }
 
-export class ProgressCircle extends React.PureComponent<ProgressCircleProps> {
-  static readonly defaultProps: Partial<ProgressCircleProps> = {
-    size: 'medium',
-  };
-
-  render() {
-    const { error } = this.props;
-
-    return error ? this.renderError() : this.renderCircle();
-  }
-
-  getDimensions() {
-    const { size } = this.props;
-
-    return size ? CIRCLE_DIMENSIONS[size] : CIRCLE_DIMENSIONS.medium;
-  }
-
-  renderCircle() {
-    const { percent, size } = this.props;
+export const ProgressCircle: React.FC<ProgressCircleProps> = typedMemo(({ error, percent, size = 'medium' }) => {
+  const getDimensions = () => (size ? CIRCLE_DIMENSIONS[size] : CIRCLE_DIMENSIONS.medium);
```

Comment:
Could it be `null`? if not you could use just `CIRCLE_DIMENSIONS[size]` because y default size is `medium`

Reply:
We could probably get rid of `getDimensinos` and just do 
```ts
const dimensions = CIRCLE_DIMENSIONS[size];
``` 

Since it will re-render if `size` changes we don't need to call a fn anymore.